### PR TITLE
Only display valid links on detail page

### PR DESCRIPTION
**Pull request by leb2dg**
Also fix the api url.

### CODE REVIEW

File: app/components/curate-work/component.js [hunk 1]
```diff
@@ -40,6 +40,21 @@ export default Ember.Component.extend({
...
     }.property('toMerge.[]'),
 
+    validLinks: Ember.computed('work.links', function() {
+        let links = this.get('work.links').content.currentState;
+
+        // http://stackoverflow.com/questions/3809401/what-is-a-good-regular-expression-to-match-a-url
+        let urlPattern = /[-a-zA-Z0-9@:%_\+.~#?&//=]{2,256}\.[a-z]{2,4}\b(\/[-a-zA-Z0-9@:%_\+.~#?&//=]*)?/gi;
+        let urlRegex = new RegExp(urlPattern);
+
+        return links.map(function(link) {
```
Comment (chrisseto) [inline app/components/curate-work/component.js]: You want to be using `.filter` here.

File: app/components/curate-work/component.js [hunk 2]
```diff
@@ -40,6 +40,21 @@ export default Ember.Component.extend({
         }];
     }.property('toMerge.[]'),
 
+    validLinks: Ember.computed('work.links', function() {
+        let links = this.get('work.links').content.currentState;
```
Comment (chrisseto) [inline app/components/curate-work/component.js]: Ember *should* take care of this for you. Using `this.get('work.links')` should be sufficient.

File: app/components/curate-work/component.js [hunk 3]
```diff
@@ -40,6 +40,21 @@ export default Ember.Component.extend({
...
+
+        // http://stackoverflow.com/questions/3809401/what-is-a-good-regular-expression-to-match-a-url
+        let urlPattern = /[-a-zA-Z0-9@:%_\+.~#?&//=]{2,256}\.[a-z]{2,4}\b(\/[-a-zA-Z0-9@:%_\+.~#?&//=]*)?/gi;
```
Comment (chrisseto) [inline app/components/curate-work/component.js]: This should be a constant at the beginning of the file.

**Comment by leb2dg** [comment]
@chrisseto, changes addressed 🍔

**Comment by aaxelb** [comment]
I realize this is hacky, but could we hide all the `http://id.crossref.org/prefix/...` and `http://id.crossref.org/member/...` links, too? They're not technically invalid but they don't lead anywhere or identify anything useful.

**Pull request closed without merging**